pkg/rpc/client/helper.go

```diff
@@ -97,33 +106,195 @@ func topMapFromStack(st []stackitem.Item) (*stackitem.Map, error) {
+	emit.Opcodes(script.BinWriter, opcode.STLOC0,                           // Pop the result of System.Contract.Call (the iterator) from estack and store it inside the 0-th cell of the local slot.
+		opcode.NEWARRAY0, // Push new empty array to estack. This array will store iterator's elements.
+		opcode.STLOC1)    // Pop the empty array from estack and store it inside the 1-th cell of the local slot.
+
+	// Start the iterator traversal cycle.
+	iteratorTraverseCycleStartOffset := script.Len()
+	emit.Opcodes(script.BinWriter, opcode.LDLOC0)                   // Load iterator from the 0-th cell of the local slot and push it on estack.
+	emit.Syscall(script.BinWriter, interopnames.SystemIteratorNext) // Call System.Iterator.Next, it will pop the iterator from estack and push `true` or `false` to estack.
+	jmpIfNotOffset := script.Len()
+	emit.Instruction(script.BinWriter, opcode.JMPIFNOT, // Pop boolean value (from the previous step) from estack, if `false`, then iterator has no more items => jump to the end of program.
+		[]byte{
+			0x00, // jump to loadResultOffset, but we'll fill this byte after script creation.
+		})
+	emit.Opcodes(script.BinWriter, opcode.LDLOC1, // Load the resulting array from 1-th cell of local slot and push it to estack.
+		opcode.LDLOC0) // Load iterator from the 0-th cell of local slot and push it to estack.
+	emit.Syscall(script.BinWriter, interopnames.SystemIteratorValue) // Call System.Iterator.Value, it will pop the iterator from estack and push its current value to estack.
+	emit.Opcodes(script.BinWriter, opcode.APPEND)                    // Pop iterator value and the resulting array from estack. Append value to the resulting array. Array is a reference type, thus, value stored at the 1-th cell of local slot will also be updated.
+	jmpOffset := script.Len()
+	emit.Instruction(script.BinWriter, opcode.JMP, // Jump to the start of iterator traverse cycle.
+		[]byte{
+			uint8(iteratorTraverseCycleStartOffset - jmpOffset), // jump to iteratorTraverseCycleStartOffset; offset is relative to JMP position.
+		})
+
+	// End of the program: push the result on stack and return.
+	loadResultOffset := script.Len()
+	emit.Opcodes(script.BinWriter, opcode.LDLOC1, // Load the resulting array from 1-th cell of local slot and push it to estack.
+		opcode.RET) // Return.
+	if err := script.Err; err != nil {
+		return nil, fmt.Errorf("failed to build iterator unwrapper script: %w", err)
+	}
+
+	// Fill in JMPIFNOT instruction parameter.
+	bytes := script.Bytes()
+	bytes[jmpIfNotOffset+1] = uint8(loadResultOffset - jmpIfNotOffset) // +1 is for JMPIFNOT itself; offset is relative to JMPIFNOT position.
+	return bytes, nil
+}
+
+// expandArrayIntoScript pushes all smartcontract.Parameter parameters from the given array
+// into the given buffer in the reverse order.
+func expandArrayIntoScript(script *io.BinWriter, slice []smartcontract.Parameter) error {
```

Comment:
Because this looks very much like `request.ExpandArrayIntoScript`. We can't use it directly here, but `emit.Array` should solve that.

Reply:
Fixed.